### PR TITLE
Mount shared

### LINKED ISSUE #3965: Unable to mount root of a shared folder
Steps to reproduce:

- Create a share
- Mount the share itself (by selecting "Use this folder")
- Observe that the folder is not mounted

Note that this only happens when you use the root of the mount, it does not occur when you select any subfolder of the share.

### PR DESCRIPTION
**Pull request by hschu12**
Added support for mounting shares as root.  Fixes #3965 . 
Also a small change to mount path to better reflect frontend names for Home and Member Files.

### CODE REVIEW

File: backend/app-orchestrator-service/src/main/kotlin/services/JobVerificationService.kt [hunk 1]
```diff
@@ -81,7 +87,9 @@ class JobVerificationService(
         // Check mounts
         run {
             val mounts = resources.filterIsInstance<AppParameterValue.File>()
+            println(mounts)
             newResources.addAll(checkAndReturnValidFiles(actorAndProject, mounts))
+            println(newResources)
```
Comment (Tiggles) [inline backend/app-orchestrator-service/src/main/kotlin/services/JobVerificationService.kt:92]:
```suggestion
```

File: backend/app-orchestrator-service/src/main/kotlin/services/JobVerificationService.kt [hunk 2]
```diff
@@ -126,26 +132,63 @@ class JobVerificationService(
...
             it.path.split("/").last()
         }.toSet()
-        if (files.size != filenames.size) {
+        if (translatedFiles.size != filenames.size) {
             throw RPCException("Cannot mount files with same name", HttpStatusCode.BadRequest)
         }
 
-        for (file in files) {
+        for (file in translatedFiles) {
+            println("file: $file")
```
Comment (Tiggles) [inline backend/app-orchestrator-service/src/main/kotlin/services/JobVerificationService.kt:191]:
```suggestion
```

**Comment by DanThrane** [comment]
I have pushed an additional commit. It should take of the bug related to mounting the root of non-shares, but I haven't tested it yet. The arg-builder bug still needs to be resolved.

File: backend/file-orchestrator-service/api/build.gradle.kts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 dependencies {
    api(project(":accounting-service:api"))
-   // api("com.github.java-json-tools:json-schema-validator:2.2.14")
+    implementation(project(mapOf("path" to ":service-lib-server")))
```
[the remark this answers is not in the document]
Comment (hschu12) [reply backend/file-orchestrator-service/api/build.gradle.kts]: Automatic import when tries to use metadata service. Forgot to remove.

File: provider-integration/integration-module/src/main/kotlin/plugins/compute/ucloud/FeatureParameter.kt [hunk 4]
```diff
@@ -127,7 +127,13 @@ private class OurArgBuilder(
                     if (components.isEmpty()) {
                         return ArgumentBuilder.Default.build(parameter, value)
                     }
-                    joinPath("/work", components[components.lastIndex]).removeSuffix("/")
+                    if (components.size == 4 && components[2] == "home") {
```
Comment (DanThrane) [inline provider-integration/integration-module/src/main/kotlin/plugins/compute/ucloud/FeatureParameter.kt:130]:
```suggestion
                    if (components.size == 4 && components[2] == PathConverter.HOME_DIRECTORY) {
```

File: provider-integration/integration-module/src/main/kotlin/plugins/compute/ucloud/FeatureParameter.kt [hunk 5]
```diff
@@ -127,7 +127,13 @@ private class OurArgBuilder(
...
+                        joinPath("/work", "Home")
+                    } else if (components.size == 6 && components[4] == "Members' Files"){
```
Comment (DanThrane) [inline provider-integration/integration-module/src/main/kotlin/plugins/compute/ucloud/FeatureParameter.kt:132]:
```suggestion
                    } else if (components.size == 6 && components[4] == PathConverter.PERSONAL_REPOSITORY){
```

**Review by DanThrane — CHANGES_REQUESTED** [review]
Minor changes

**Review by DanThrane — APPROVED** [review]

**Pull request merged**